pkg/webhook/policy.go

```diff
@@ -155,8 +157,26 @@ func (h *validationHandler) Handle(ctx context.Context, req admission.Request) a
 		}
 	}()
 
+	var obj runtime.Object
+	var scope conversion.Scope
```

Comment:
This should work (modulo error handling):

```
obj := &unstructured.Unstructured{}
runtimeScheme.Convert(&req.AdmissionRequest.Object, obj, nil)
```

Otherwise you are invoking the hidden conversion functions (those names with underscores), directly, which may break if scheme changes how it works

Reply:
this didn't work but looks like `deserializer.Decode` works

Reply:
Interesting AFAIK, they are both initialized via scheme, but maybe they use a different scheme.

What does the deserializer.Decode() code look like?

Reply:
looks like this is the code for unstructured decode: https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/unstructured/helpers.go#L327

we already use in few places like https://github.com/open-policy-agent/gatekeeper/blob/f3ec64a56154cd991e98387329083d243c967879/pkg/webhook/policy.go#L201

Reply:
thanks for finding the source code!